File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> queue = new LinkedList<>();
        
        for (int i=0; i<progresses.length; i++) {
            int days = (int)((100-progresses[i]) / speeds[i]);
            days += ((100-progresses[i]) % speeds[i]==0) ? 0 : 1;
            queue.offer(days);
        }
        
        while(!queue.isEmpty()) {
            int cnt = 1;
            int days = queue.poll();
            while (!queue.isEmpty() && days >= queue.peek()) {
                cnt++;
                queue.poll();
            }
            answer.add(cnt);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    i = 0
    while i < len(progresses):
        days =  (100 - progresses[i] + speeds[i] - 1) // speeds[i]
        cnt = 1
        while (i+1 < len(progresses) and progresses[i+1] + speeds[i+1]* days >= 100):
            i += 1
            cnt += 1
        i+=1
        answer.append(cnt)
    return answer